react-hot-button-component/src/hot-button.jsx

```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+
+class HotButton extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { clicks: 0 };
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  handleClick() {
+    this.setState({ clicks: this.state.clicks + 1 });
+  }
+
+  render() {
+    const clicks = this.state.clicks;
+
+    if (clicks < 4) {
+      const className = 'hot-btn-purple';
+      return (
+        <div>
+          <button onClick={this.handleClick} className={className}>Hot Button</button>
+        </div>
+      );
+    } else if (clicks < 7) {
+      const className = 'hot-btn-light-purple';
+      return (
+        <button onClick={this.handleClick} className={className}>Hot Button</button>
+      );
+    } else if (clicks < 10) {
+      const className = 'hot-btn-red';
+      return (
+        <button onClick={this.handleClick} className={className}>Hot Button</button>
+      );
+    } else if (clicks < 13) {
+      const className = 'hot-btn-orange';
+      return (
+        <button onClick={this.handleClick} className={className}>Hot Button</button>
+      );
+    } else if (clicks < 16) {
+      const className = 'hot-btn-yellow';
+      return (
+        <button onClick={this.handleClick} className={className}>Hot Button</button>
+      );
+    } else {
+      const className = 'hot-btn-white';
+      return (
+        <button onClick={this.handleClick} className={className}>Hot Button</button>
+      );
+    }
+  }
```

Comment:
This is a lot of repetition and would be hard to maintain. You are half way doing it the correct way by saving the class into a variable but the if statements should just determine the class name then at the end have a single return.
```JSX
const clicks = this.state.clicks;
let className = '';

if (clicks < 4) {
  className = 'hot-btn-purple';
} else if (clicks < 7) {
  className = 'hot-btn-light-purple';
} else if (clicks < 10) {
  className = 'hot-btn-red';
} else if (clicks < 13) {
  className = 'hot-btn-orange';
} else if (clicks < 16) {
  className = 'hot-btn-yellow';
} else {
  className = 'hot-btn-white';
}

return (
  <button onClick={this.handleClick} className={className}>
    Hot Button
  </button>
);
```